serchProc.js

```diff
@@ -0,0 +1,122 @@
+            console.log(responseData);
+            //list를 출력하는 for
+            for (var i = 0; i <= responseData.length; i++) {
+                storeName = responseData[i].storeName;
+                storeNumber = responseData[i].storeNumber;
+                id = responseData[i]._id;
+                roadName = responseData[i].roadName;
+                storeX = responseData[i].storeX;
+                storeY = responseData[i].storeY;
+                writer = responseData[i].writer;
+                menu = responseData[i].menu;
+
+                const button = `<button id='btn' onclick='showData("${id}", "${storeName}", "${storeNumber}", "${roadName}", "${storeX}")'>정보보기</button>`;
+                document.getElementById("resultList").innerHTML += num + "." + storeName + storeNumber+ button + "<br>";
+                //document.getElementById("resultList").innerHTML += button + "<br>";
+                num++;
+            }
+        }
+    }
+    xhr.send(null);
+}
+getData()
+
+//db를 검색하는 함수
+function serchDb() {
+    const type = document.getElementById('serchType').value;
+    const sKey = document.getElementById('serching').value;
+
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', "/api/storeDB/serchKey?serchId=" + sKey + "&type=" + type, true);
+    xhr.onreadystatechange = function () { // Call a function when the state changes.
+        console.log('this.readyState: ', this.readyState);
+        console.log('this.state:', this.state);
+
+        if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+            document.getElementById('resultList').innerHTML = "";
+
+            // Request finished. Do processing here.
+            const responseList = JSON.parse(this.response);
+            console.log(responseList.length);
+            console.log(responseList);
+            //clearDiv();
+            if (responseList.length == 0) {
+                alert("검색결과가 존재하지 않습니다.!!!");
+            } else {
+                for (var i = 0; i <= responseList.length; i++) {
+
+                    storeName = responseList[i].storeName;
+                    storeNumber = responseList[i].storeNumber;
+                    roadName = responseList[i].roadName;
+                    storeX = responseList[i].storeX;
+                    storeY = responseList[i].storeY;
+                    id = responseList[i]._id;
+                    writer = responseList[i].writer;
+                    menu = responseList[i].menu;
+
+                    const button = `<button id='inputb' onclick='showData("${storeName}", "${storeNumber}", "${roadName}", "${storeX}")'>정보보기</button>`;
+                    document.getElementById("resultList").innerHTML += (i + 1) + "." + storeName + storeNumber;
+                    document.getElementById("resultList").innerHTML += button + "<br>";
+                }
+            }
+        }
+
+    }
+    xhr.send(null);
+}
+
+function showData(storeName, storeNumber, roadName, storeX, storeY, id) {
```

Comment:
```suggestion
function showData(storeName, storeNumber, roadName) {
```
storeX, storeY, id 매개변수는 사용하지 않으므로 지워도 될 것 같습니다.